packages/app/features/top/ChannelScreen.tsx

```diff
@@ -247,6 +247,16 @@ export default function ChannelScreen({
     },
     [navigateToUserProfile]
   );
+  const loadNewerUnlessAlreadyLoading = useCallback(() => {
+    if (!isLoadingPosts) {
+      loadNewer();
+    }
+  }, [loadNewer, isLoadingPosts]);
+  const loadOlderUnlessAlreadyLoading = useCallback(() => {
+    if (!isLoadingPosts) {
+      loadOlder();
+    }
```

Comment:
i chose not to queue loads here – if we queued loads, then we'd have close to the same behavior as _not_ guarding on `isLoadingPosts`:
1. user scrolls to bottom of list
2. load starts
3. user is still at bottom of list (nudges scroll maybe)
4. `onScrollEndReached` called a second time – we queue the request to load more
5. load from step 2 completes; we immediately start the queued load, even though the user already has posts in that direction

With the current simple implementation,
1. user scrolls to bottom of list
2. load starts
3. user is still at bottom of list (nudges scroll maybe)
4. `onScrollEndReached` called a second time – we discard the request to load more
5. load from step 2 completes, user has items
and worse case:
1. user scrolls to top of list
2. load starts for older posts
3. user changes mind, scrolls to bottom of list
4. load request for newer posts is discarded since load is active
5. load from step 2 completes, but they are not in user's view
6. if user still wants newer posts, they'll probably be rage-scrolling down :/ so they'll eventually get the posts

not great, but i think workable for now?

Reply:
Make sense, definitely workable for now. 

I'd expect worst case direction switch is somewhat infrequent. Anecdotally, I only really do that if the first load is super slow.